lib/statsd/instrument/strict.rb

```diff
@@ -97,7 +97,47 @@ def collect_metric(type, name, value, sample_rate:, tags: nil, prefix:, metadata
         nil # We explicitly discard the return value, so people cannot depend on it.
       end
     end
+
+    module StrictMetaprgramming
+      def statsd_measure(method, name, sample_rate: nil, tags: nil,
+        prefix: StatsD.prefix, no_prefix: false, as_dist: false)
+
+        check_method_and_metric_name(method, name)
+        super
+      end
+
+      def statsd_distribution(method, name, sample_rate: nil, tags: nil, prefix: StatsD.prefix, no_prefix: false)
+        check_method_and_metric_name(method, name)
+        super
+      end
+
+      def statsd_count_success(method, name, sample_rate: nil, tags: nil, prefix: StatsD.prefix, no_prefix: false)
+        check_method_and_metric_name(method, name)
+        super
+      end
+
+      def statsd_count_if(method, name, sample_rate: nil, tags: nil, prefix: StatsD.prefix, no_prefix: false)
+        check_method_and_metric_name(method, name)
+        super
+      end
+
+      def statsd_count(method, name, sample_rate: nil, tags: nil, prefix: StatsD.prefix, no_prefix: false)
+        check_method_and_metric_name(method, name)
+        super
+      end
+
+      private
+
+      def check_method_and_metric_name(method, metric_name)
+        raise ArgumentError, "The method name should be provided as symbol" unless method.is_a?(Symbol)
```

Comment:
Could be interested to say what you got instead, might make debugging much easier some times.